secure/json-rpc/src/lib.rs

```diff
@@ -63,18 +63,71 @@ impl From<FromHexError> for Error {
     }
 }
 
+/// The interface any Json Rpc client implementation should provide.
+pub trait JsonRpcClient {
+    /// Submits a signed transaction to the Libra blockchain via the JSON RPC API.
+    fn submit_signed_transaction(&self, signed_transaction: SignedTransaction)
+        -> Result<(), Error>;
+
+    /// Returns the associated AccountState for a specific account at a given version height
+    /// using the JSON RCP API. If no version height is specified, the latest is returned by the
+    /// JSON RPC server.
+    fn get_account_state(
+        &self,
+        account: AccountAddress,
+        version: Option<u64>,
+    ) -> Result<AccountState, Error>;
+}
+
 /// Provides a lightweight JsonRpcClient implementation.
-pub struct JsonRpcClient {
+pub struct LightweightJsonRpcClient {
     host: String,
 }
 
-impl JsonRpcClient {
+impl LightweightJsonRpcClient {
     pub fn new(host: String) -> Self {
         Self { host }
     }
 
-    /// Submits a signed transaction to the Libra blockchain via the JSON RPC API.
-    pub fn submit_signed_transaction(
+    // Executes the specified request method using the given parameters by contacting the JSON RPC
+    // server.
+    fn execute_request(&self, method: String, params: Vec<Value>) -> Response {
+        ureq::post(&self.host)
+            .timeout_connect(REQUEST_TIMEOUT)
+            .send_json(
+                json!({"jsonrpc": JSON_RPC_VERSION, "method": method, "params": params, "id": 0}),
+            )
+    }
+
+    // Sends a submit() request to the JSON RPC server using the given transaction.
+    fn submit_transaction(&self, signed_transaction: SignedTransaction) -> Result<Response, Error> {
+        let method = "submit".to_string();
+        let params = vec![Value::String(hex::encode(lcs::to_bytes(
+            &signed_transaction,
+        )?))];
+        Ok(self.execute_request(method, params))
+    }
+
+    // Sends a get_account_state_with_proof() request to the JSON RPC server for the specified
+    // account, optional version height and optional ledger_version height (for the proof).
+    fn get_account_state_with_proof(
+        &self,
+        account: AccountAddress,
+        version: Option<u64>,
+        ledger_version: Option<u64>,
+    ) -> Result<Response, Error> {
+        let method = "get_account_state_with_proof".to_string();
```

Comment:
use into

Reply:
Done (and changed all other to_string() to into(), where appropriate). Thanks!